lib/oauth.js

```diff
@@ -354,32 +344,50 @@ internals.Client.prototype._prepare = function (method, uri, params, oauth, opti
     oauth.oauth_signature_method = 'HMAC-SHA1';
     oauth.oauth_signature = this.signature(method, uri, params, oauth, options.secret);
 
-    var query = params ? ('?' + Querystring.encode(params)) : '';
-    return next(method, uri + query, internals.Client.header(oauth));
-};
+    if (this.settings.version) {
+        oauth.oauth_version = this.settings.version;
+    }
 
+    // Calculate OAuth header
 
-internals.Client.prototype.request = function (method, uri, params, oauth, tokenSecret, type, callback) {
+    var requestOptions = {
+        headers: {
+            Authorization: internals.Client.header(oauth)
+        }
+    };
 
-    var desc = this.provider + ' ' + type;
+    if (params) {
+        var paramsString = internals.queryString(params);
+        if (method === 'get') {
+            uri += '?' + paramsString;
+        }
+        else {
+            requestOptions.payload = paramsString;
+            requestOptions.headers['content-type'] = 'application/x-www-form-urlencoded';
+        }
+    }
 
-    this._prepare(method, uri, params, oauth, { secret: tokenSecret }, function (method, uri, header) {
+    if (options.stream) {
+        return Wreck.request(method, uri, requestOptions, callback);
+    }
 
-        Wreck[method.toLowerCase()](uri, { headers: { Authorization: header } }, function (err, res, payload) {
+    var desc = (options.desc || 'resource');
```

Comment:
I know. I just like them for readability. It breaks the `=` from the rest of the statement.